packages/assets/src/utils/url/makeAbsoluteUrl.ts

```diff
@@ -18,15 +20,25 @@ let baseUrl: string;
  * If it's not, we convert it
  * @param url - The URL to test
  * @param customBaseUrl - The base URL to use
+ * @param customRootUrl - The root URL to use
  */
-export function makeAbsoluteUrl(url: string, customBaseUrl?: string): string
+export function makeAbsoluteUrl(url: string, customBaseUrl?: string, customRootUrl?: string): string
```

Comment:
This definitely needs some unit-tests to handle the different permutations of platforms and requests.